### ISSUE TITLE
Replace `TORCH_CHECK` and `TORCH_INTERNAL_ASSERT` Macros

### ISSUE THREAD

**Issue by kevinstephano** [NVIDIA/Fuser#668]
`TORCH_INTERNAL_ASSERT` asks the user to report errors to Pytorch which no longer makes sense.  Additionally, it would be nice if the macros automatically gave a function signature and line number in a file so the user can find the error faster.

One example where that would be nice is in `arith.cpp` for the `size()` function where the user manually adds the `__FUNCTION__` macro.

```
Val* size(TensorView* inp, int64_t dim) {
  auto iter_domains = TensorDomain::noReductions(inp->getMaybeRFactorDomain());
  auto idx = dim;
  if (idx < 0) {
    idx = static_cast<int64_t>(iter_domains.size()) + idx;
  }
  TORCH_CHECK(
      (idx >= 0) && (static_cast<size_t>(idx) < iter_domains.size()),
      __FUNCTION__,
      ": The dimension requested is beyond the bounds of the shape of the indexed tensor!",
      " Tensor Dims: ",
      iter_domains.size(),
      " Dim: ",
      dim);
  return iter_domains.at(idx)->getMaybeExpandedExtent();
}
```

**Comment by jjsjann123** [comment]
I know we still have a few remaining issues that we want to tackle. (c10::irange and backtrace?!?!). Do you want to open another issue to track those? @Priya2698

**Comment by Priya2698** [comment]
The implementation in PR #811 includes the backtrace details in the macros. I can open another issue for `c10::irange`, but that is used within several other files. Do we want to completely avoid using `c10`? In that case, we can open an issue to gradually remove it.

**Comment by jjsjann123** [comment]
sorry I wasn't thinking straight when I typed and forgot that backtrace stuff is already migrated.

Kevin's long-term vision is to not depend on c10 stuff, so let's just make a list of remaining things to remove in an issue and carry it through eventually.
